exec.c

```diff
@@ -26,8 +26,93 @@
 #include "common.h"
 #include "exec.h"
 #include "malloc.h"
+#include "str.h"
 #include <unistd.h>
 
+
+/* Duplicates a string array adding extra spaces,
+ * returns the new array that must be free'd
+ */
+static char **
+duplicate_string_array(const char **array, int extra_space, int *count)
+{
+	const char **s;
+	char **ret;
+
+	*count = extra_space + 1;
+
+	if (array)
+		for (s = array; *s; s++)
+			(*count)++;
+
+
+	/* relying in the fact that calloc initializes memory to zero */
+	ret = (char **) real_calloc(*count, sizeof(char *));
+
+	if (ret) {
+		int i;
+
+		for (i = 0; i < (*count) - extra_space; i++)
+			ret[i] = (char *) array[i];
```

Comment:
I would need a foor loop to set the remaining four spots to zero. In reality calloc is pretty fast due to OS specific tricks (https://stackoverflow.com/questions/2688466/why-mallocmemset-is-slower-than-calloc)